MediaBrowser.Providers/Manager/ItemImageProvider.cs

```diff
@@ -73,31 +87,42 @@ public bool ValidateImages(BaseItem item, IEnumerable<IImageProvider> providers,
             return hasChanges;
         }
 
+        /// <summary>
+        /// Refreshes from the providers according to the given options.
+        /// </summary>
+        /// <param name="item">The <see cref="BaseItem"/> to gather images for.</param>
+        /// <param name="libraryOptions">The library options.</param>
+        /// <param name="providers">The providers to query for images.</param>
+        /// <param name="refreshOptions">The refresh options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The refresh result.</returns>
         public async Task<RefreshResult> RefreshImages(
             BaseItem item,
             LibraryOptions libraryOptions,
             List<IImageProvider> providers,
             ImageRefreshOptions refreshOptions,
             CancellationToken cancellationToken)
         {
+            List<ItemImageInfo> oldBackdropImages = new List<ItemImageInfo>();
```

Comment:
You don't seem to add anything to the list, so just use an Array and save an allocation (hint: `Array.Empty<ItemImageInfo>() is free).

Reply:
It may be free to create an empty array, but BaseItem.RemoveImages (where this is used in PruneImages) requires a List and it seemed simpler and more readable to stick with a List type all the way through the handling of this.

Alternately, if you want me to change the `BaseItem.RemoveImages(List<ItemImageInfo> deletedImages)` call to `BaseItem.RemoveImages(IEnumerable<ItemImageInfo> deletedImages)` and use array throughout I can, I just was hesitant to mess with the interface there.

Reply:
The only usage in the server for `BaseItem.RemoveImages` is in `ItemImageProvider`, so change away. It also shouldn't break any plugins if they were to use it as well as `IEnumerable` is implemented by IList